### ISSUE TITLE
SchemaSpy with Informix 12.10 FC10: "cursor has been previously released" exception. SchemaSpy crashes

### ISSUE THREAD

**Issue by begooden-it** [schemaspy/schemaspy#424]
When launching schemaSpy against Informix 12.10 FC10, I get always the following exception
ERROR - The cursor has been previously released and is unavailable.
.SQLException: The cursor has been previously released and is unavailable.

and schemaspy crashes

## Expected Behavior
With other versions, I never had this exception on Informix

## Steps to Reproduce (for bugs)
java -jar "C:\Tools\SchemaSpy\schemaSpy.jar" -dp "C:\Informix\1210FC9\jdbc\lib\ifxjdbc.jar" -t informix -db kandoodb -host 192.168.1.20:29140 -port 29140 -u informix -p informix -server production_tcp -o kandoodb -connprops="CLIENT_LOCALE\=en_US.utf8;DB_LOCALE\=en_US.57372" -charset UTF-8 -s informix

### Output
C:\Tools\SchemaSpy>java -jar "C:\Tools\SchemaSpy\schemaSpy.jar" -dp "C:\Informix\1210FC9\jdbc\lib\ifxjdbc.jar" -t informix -db kandoodb -host 192.168.1.20:29140 -por
t 29140 -u informix -p informix -server production_tcp -o kandoodb -connprops="CLIENT_LOCALE\=en_US.utf8;DB_LOCALE\=en_US.57372" -charset UTF-8 -s informix
  ____       _                          ____
 / ___|  ___| |__   ___ _ __ ___   __ _/ ___| _ __  _   _
 \___ \ / __| '_ \ / _ \ '_ ` _ \ / _` \___ \| '_ \| | | |
  ___) | (__| | | |  __/ | | | | | (_| |___) | |_) | |_| |
 |____/ \___|_| |_|\___|_| |_| |_|\__,_|____/| .__/ \__, |
                                             |_|    |___/

                                              6.0.0-rc2

SchemaSpy generates an HTML representation of a database schema's relationships.

INFO  - Starting Main v6.0.0-rc2 on C2RSI-PC with PID 7104 (C:\Tools\SchemaSpy\schemaspy.jar started by begooden-it in C:\Tools\SchemaSpy)
INFO  - No active profile set, falling back to default profiles: default
INFO  - Found configuration file: schemaspy.properties
INFO  - Starting schema analysis
INFO  - Resolving dbType: informix ->
        /C:/Tools/SchemaSpy/schemaspy.jar!/BOOT-INF/classes!/org/schemaspy/types/informix.properties
INFO  - Connected to Informix Dynamic Server - 12.10.FC10
INFO  - Gathering schema details
Gathering schema details.............................................................................................................................................
.....................................................................................................................................................................
.....................................................................................................................................................................
.....................................................................................................................................................................
.........................................................................................WARN  - Unable to extract the number of rows for table x_customer, using '-1
'
WARN  - java.sql.SQLException: Reached maximum error limit during load: (10).
............(36sec)
Connecting relationships.............................................................................................................................................
.....................................................................................................................................................................
........................................ERROR - The cursor has been previously released and is unavailable.
java.sql.SQLException: The cursor has been previously released and is unavailable.
        at com.informix.util.IfxErrMsg.buildException(IfxErrMsg.java:474)
        at com.informix.util.IfxErrMsg.getSQLException(IfxErrMsg.java:443)
        at com.informix.util.IfxErrMsg.getSQLException(IfxErrMsg.java:394)
        at com.informix.jdbc.IfxSqli.addException(IfxSqli.java:3191)
        at com.informix.jdbc.IfxSqli.receiveError(IfxSqli.java:3472)
        at com.informix.jdbc.IfxSqli.dispatchMsg(IfxSqli.java:2379)
        at com.informix.jdbc.IfxSqli.receiveMessage(IfxSqli.java:2304)
        at com.informix.jdbc.IfxSqli.executeFetch(IfxSqli.java:2038)
        at com.informix.jdbc.IfxSqli.getaRow(IfxSqli.java:4082)
        at com.informix.jdbc.IfxResultSet.next(IfxResultSet.java:494)
        at com.informix.jdbc.IfxDatabaseMetaData.getImportedKeys(IfxDatabaseMetaData.java:5389)
        at org.schemaspy.service.TableService.connectForeignKeys(TableService.java:203)
        at org.schemaspy.service.DatabaseService.connectTables(DatabaseService.java:261)
        at org.schemaspy.service.DatabaseService.gatheringSchemaDetails(DatabaseService.java:67)
        at org.schemaspy.SchemaAnalyzer.analyze(SchemaAnalyzer.java:205)
        at org.schemaspy.SchemaAnalyzer.analyze(SchemaAnalyzer.java:101)
        at org.schemaspy.Main.runAnalyzer(Main.java:80)
        at org.schemaspy.Main.run(Main.java:73)
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:732)
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:716)
        at org.springframework.boot.SpringApplication.afterRefresh(SpringApplication.java:703)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:304)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107)
        at org.schemaspy.Main.main(Main.java:56)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: java.sql.SQLException: null
        at com.informix.util.IfxErrMsg.buildException(IfxErrMsg.java:474)
        at com.informix.util.IfxErrMsg.getSQLException(IfxErrMsg.java:443)
        at com.informix.util.IfxErrMsg.getSQLException(IfxErrMsg.java:394)
        at com.informix.jdbc.IfxSqli.receiveError(IfxSqli.java:3477)
        ... 28 common frames omitted
INFO  - Started Main in 47.609 seconds (JVM running for 48.355)

## Context

Just want to run schemaspy

* Informix
* Database driver version (is you use a special one):
* Your DDL (sample how to reproduce the issue):
* Full command including arguments how you run SchemaSpy:
java -jar "C:\Tools\SchemaSpy\schemaSpy.jar" -dp "C:\Informix\1210FC9\jdbc\lib\ifxjdbc.jar" -t informix -db kandoodb -host 192.168.1.20:29140 -port 29140 -u informix -p informix -server production_tcp -o kandoodb -connprops="CLIENT_LOCALE\=en_US.utf8;DB_LOCALE\=en_US.57372" -charset UTF-8 -s informix

## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39):
* Operating System and version (e.g. Windows 10):
Windows 7
* Java version:

**Comment by npetzall** [comment]
Could you give it a go either with snapshot or the release of 6.0.0 instead of rc2?

**Comment by begooden-it** [comment]
Hi Niels,

I have something weird here. In fact I am testing with 2 forms of distribution of this jsdk: one 'standalone' in the full windows Informix RDBMS distribution 12.10 FC9, which is the one that causes problems, and I gave a second try with Informix CSDK, which works till the end...

Please let me sort out this question before taking this a a real issue, and leave this issue on pause until I wake it up or close.

I will give a try to the snapshot too ASAP

Thanks
Eric

**Comment by npetzall** [comment]
If you run snapshot it might work. There are two issues that have been fixed that might cause this type of behaviour.

First there was an issue that when not specifying `-s` on the command-line it thought of all schemas as remote. This was partially fixed in #339. When doing that PR i encountered a problem with mssql trying to use a closed resultset. Which I believe was due to some form of recursion.

The fix in #339 introduced a bug that was corrected in #398

So I think snapshot should work, why it works with different drivers is interesting, but probably has to do with the implementation of resultset or connection.

So I'm waiting with excitement on the results for 6.0.1-SNAPSHOT.

**Comment by begooden-it** [comment]
Great !





Could you just (again) provide me with full URL of the snapshot. Not easy
to find J



Thanks bunches



Other question, clearly a new topic:

I know Postgresql and more DBMS offer out of the box the functionality of
 setting comments for tables, columns etc.
Informix does not have this functionality, but could have. I can talk with
the Informix lab to implement that, but in the meantime this can be
simulated.



What would you need to implement this functionality ?



I will create a new request (issue) for that but wanted to discuss it
before starting this process



Thanks



Eric



*From:* Nils Petzäll [mailto:notifications@github.com]
*Sent:* mercredi 12 septembre 2018 09:04
*To:* schemaspy/schemaspy <schemaspy@noreply.github.com>
*Cc:* begooden-it <eric.vercelletto@begooden-it.com>; Author <
author@noreply.github.com>
*Subject:* Re: [schemaspy/schemaspy] SchemaSpy with Informix 12.10 FC10:
"cursor has been previously released" exception. SchemaSpy crashes (#424)



If you run snapshot it might work. There are two issues that have been
fixed that might cause this type of behaviour.

First there was an issue that when not specifying -s on the command-line it
thought of all schemas as remote. This was partially fixed in #339
<https://github.com/schemaspy/schemaspy/pull/339>. When doing that PR i
encountered a problem with mssql trying to use a closed resultset. Which I
believe was due to some form of recursion.

The fix in #339 <https://github.com/schemaspy/schemaspy/pull/339>
introduced a bug that was corrected in #398
<https://github.com/schemaspy/schemaspy/pull/398>

So I think snapshot should work, why it works with different drivers is
interesting, but probably has to do with the implementation of resultset or
connection.

So I'm waiting with excitement on the results for 6.0.1-SNAPSHOT.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/schemaspy/schemaspy/issues/424#issuecomment-420535843>,
or mute the thread

**Comment by npetzall** [comment]
https://github.com/schemaspy/schemaspy/raw/gh-pages/schemaspy-6.0.1-SNAPSHOT.jar

Comments handling can be done with simple queries. So it could be implemented as a table_comments table with columns to identify the table and the comment.

**Comment by begooden-it** [comment]
Thanks Niels





So if i create a table called table_comments like this:

Tabid integer ( references systables.tabid)

Comments lvarchar(4096)



And another table called column_comments

Tabid integer

Colno smallint

Column_comments lvarchar(4096)



That would make it and you could implement it ?







*From:* Nils Petzäll [mailto:notifications@github.com]
*Sent:* mercredi 12 septembre 2018 09:52
*To:* schemaspy/schemaspy <schemaspy@noreply.github.com>
*Cc:* begooden-it <eric.vercelletto@begooden-it.com>; Author <
author@noreply.github.com>
*Subject:* Re: [schemaspy/schemaspy] SchemaSpy with Informix 12.10 FC10:
"cursor has been previously released" exception. SchemaSpy crashes (#424)



https://github.com/schemaspy/schemaspy/raw/gh-pages/schemaspy-6.0.1-SNAPSHOT.jar

Comments handling can be done with simple queries. So it could be
implemented as a table_comments table with columns to identify the table
and the comment.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/schemaspy/schemaspy/issues/424#issuecomment-420548613>,
or mute the thread

**Comment by npetzall** [comment]
Let's move the comments to a new issue and call it an enhancement. It's actually an enhancement that can be implemented without code change.

**Comment by npetzall** [comment]
@begooden-it ping

**Comment by begooden-it** [comment]
Hi Nils

first of all happy new year!
after this, very sorry for the delay I have been caught in a very exclusive assignment which left my head over water one day before christmas!

FYI, I have moved schemaSpy on Centos OS, on which I feel more at ease.
SO I have been testing again the cursor issue, and it keeps happening on an irregular basis. I have attached the exception message

the version is 6.0.0-rc2

I can try with IFMX 12.10 FC12 in a few days

Could you find anything ?

Regards
Eric

**Comment by npetzall** [comment]
We both have final 6.0.0 release and a 6.1.0-SNAPSHOT.

Would be greate if you could test the 6.1.0-SNAPSHOT. I think I've fixed an issue earlier that might relate to the issue.

**Comment by begooden-it** [comment]
Ok got it



I’m testing, and could not reproduce the cursor error until now



But I have bunches of this error



ERROR - dot -Tpng:cairo
/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied1degrees.dot
-o/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied1degrees.png
-Tcmapx: in label of node company

ERROR - dot -Tpng:cairo
/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied2degrees.dot
-o/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied2degrees.png
-Tcmapx: Warning: cell size too small for content

ERROR - dot -Tpng:cairo
/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied2degrees.dot
-o/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied2degrees.png
-Tcmapx: in label of node company

ERROR - dot -Tpng:cairo
/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied2degrees.dot
-o/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied2degrees.png
-Tcmapx: Warning: cell size too small for content



What can I do



This is centos 7.5

Graphviz is v 2.30





I have been trying to install graphviz 2.40 but on Centos OS there is no
yum package, and with the rpm package, I have almost been kill my yum
database

Do I need graphviz > .2.30 as the GA version says ?



Thanks

Eric

*De :* Nils Petzäll [mailto:notifications@github.com]
*Envoyé :* mardi 8 janvier 2019 17:30
*À :* schemaspy/schemaspy <schemaspy@noreply.github.com>
*Cc :* begooden-it <eric.vercelletto@begooden-it.com>; Mention <
mention@noreply.github.com>
*Objet :* Re: [schemaspy/schemaspy] SchemaSpy with Informix 12.10 FC10:
"cursor has been previously released" exception. SchemaSpy crashes (#424)



It's in the readme

https://github.com/schemaspy/schemaspy/raw/gh-pages/schemaspy-6.1.0-SNAPSHOT.jar

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/schemaspy/schemaspy/issues/424#issuecomment-452362488>,
or mute the thread

**Comment by npetzall** [comment]
There's really just one bad version which is 2.31 so I think 2.30 should be OK. It's output as error in schemaspy since it's written to stderr of graphviz. So they might all just be warning. Check the output is my advice.

If it's 6.1.0-SNAPSHOT you can use the `-vizjs` argument and don't need graphviz to be installed it will use javascript version. Depending on database size it might work. I have created two issue where db is to large to work.

**Comment by npetzall** [comment]
You could check the content of `/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied1degrees.dot` something with label in company. If it's not secret or confidential you could post the content and I can have look.

For the cell size, I wonder if there is a row content that we don't take into account. It's probably the longest. I'll glance over the code for the cell thing.

**Comment by begooden-it** [comment]
No secret here :

Here we go



Does the snapshot include the patch you made for the table comments and
column comments?





digraph "oneDegreeRelationshipsDiagramImplied" {

  graph [

    rankdir="RL"

    bgcolor="#ffffff"

    label="\nGenerated by SchemaSpy"

    labeljust="l"

    nodesep="0.18"

    ranksep="0.46"

    fontname="Helvetica"

    fontsize="11"

    ration="compress"

  ];

  node [

    fontname="Helvetica"

    fontsize="11"

    shape="plaintext"

  ];

  edge [

    arrowsize="0.8"

  ];

  "reqbackord":"cmpy_code":w -> "company":"cmpy_code":e [arrowhead=none
dir=back arrowtail=crowodot style=dashed];

  "company" [

   label=<

    <TABLE BORDER="0" CELLBORDER="1" CELLSPACING="0" BGCOLOR="#ffffff">

      <TR><TD COLSPAN="3"  BGCOLOR="#f5f5f5"><TABLE BORDER="0"
CELLSPACING="0"><TR><TD ALIGN="LEFT"><B>company</B></TD><TD
ALIGN="RIGHT">[table]</TD></TR></TABLE></TD></TR>

      <TR><TD PORT="cmpy_code" COLSPAN="3" ALIGN="LEFT"><TABLE BORDER="0"
CELLSPACING="0" ALIGN="LEFT"><TR ALIGN="LEFT"><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="15" HEIGHT="16"><IMG
SRC="/home/StagingArea/schemaSpy/kandoodb/images/primaryKeys.png"/></TD><TD
ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="100"
HEIGHT="16">cmpy_code</TD></TR></TABLE></TD></TR>

      <TR><TD PORT="elipses" COLSPAN="3" ALIGN="LEFT">...</TD></TR>

      <TR><TD ALIGN="LEFT" BGCOLOR="#ffffff">  </TD><TD ALIGN="RIGHT"
BGCOLOR="#ffffff">1 row</TD><TD ALIGN="RIGHT" BGCOLOR="#ffffff">546
&gt;</TD></TR>

    </TABLE>>

    URL="company.html"

    target="_top"

    tooltip="company"

  ];

  "reqbackord" [

   label=<

    <TABLE BORDER="2" CELLBORDER="1" CELLSPACING="0" BGCOLOR="#ffffff">

      <TR><TD COLSPAN="4"  BGCOLOR="#f5f5f5"><TABLE BORDER="0"
CELLSPACING="0"><TR><TD ALIGN="LEFT"><B>reqbackord</B></TD><TD
ALIGN="RIGHT">[table]</TD></TR></TABLE></TD></TR>

      <TR><TD PORT="cmpy_code" COLSPAN="2" BGCOLOR="#ffffff"
ALIGN="LEFT"><TABLE BORDER="0" CELLSPACING="0" ALIGN="LEFT"><TR
ALIGN="LEFT"><TD ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="15" HEIGHT="16"><IMG
SRC="/home/StagingArea/schemaSpy/kandoodb/images/foreignKeys.png"/></TD><TD
ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="84"
HEIGHT="16">cmpy_code</TD></TR></TABLE></TD><TD PORT="cmpy_code.type"
ALIGN="LEFT">char[2]</TD></TR>

      <TR><TD PORT="part_code" COLSPAN="2" BGCOLOR="#ffffff"
ALIGN="LEFT"><TABLE BORDER="0" CELLSPACING="0" ALIGN="LEFT"><TR
ALIGN="LEFT"><TD ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="15"
HEIGHT="16"></TD><TD ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="84"
HEIGHT="16">part_code</TD></TR></TABLE></TD><TD PORT="part_code.type"
ALIGN="LEFT">char[15]</TD></TR>

      <TR><TD PORT="req_num" COLSPAN="2" BGCOLOR="#ffffff"
ALIGN="LEFT"><TABLE BORDER="0" CELLSPACING="0" ALIGN="LEFT"><TR
ALIGN="LEFT"><TD ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="15"
HEIGHT="16"></TD><TD ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="84"
HEIGHT="16">req_num</TD></TR></TABLE></TD><TD PORT="req_num.type"
ALIGN="LEFT">integer[10]</TD></TR>

      <TR><TD PORT="line_num" COLSPAN="2" BGCOLOR="#ffffff"
ALIGN="LEFT"><TABLE BORDER="0" CELLSPACING="0" ALIGN="LEFT"><TR
ALIGN="LEFT"><TD ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="15"
HEIGHT="16"></TD><TD ALIGN="LEFT" FIXEDSIZE="TRUE" WIDTH="84"
HEIGHT="16">line_num</TD></TR></TABLE></TD><TD PORT="line_num.type"
ALIGN="LEFT">smallint[5]</TD></TR>

      <TR><TD PORT="ware_code" COLSPAN="2" ALIGN="LEFT"><TABLE BORDER="0"
CELLSPACING="0" ALIGN="LEFT"><TR ALIGN="LEFT"><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="15" HEIGHT="16"></TD><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="84" HEIGHT="16">ware_code</TD></TR></TABLE></TD><TD
PORT="ware_code.type" ALIGN="LEFT">char[3]</TD></TR>

      <TR><TD PORT="require_qty" COLSPAN="2" ALIGN="LEFT"><TABLE BORDER="0"
CELLSPACING="0" ALIGN="LEFT"><TR ALIGN="LEFT"><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="15" HEIGHT="16"></TD><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="84"
HEIGHT="16">require_qty</TD></TR></TABLE></TD><TD PORT="require_qty.type"
ALIGN="LEFT">decimal[12,4]</TD></TR>

      <TR><TD PORT="alloc_qty" COLSPAN="2" ALIGN="LEFT"><TABLE BORDER="0"
CELLSPACING="0" ALIGN="LEFT"><TR ALIGN="LEFT"><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="15" HEIGHT="16"></TD><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="84" HEIGHT="16">alloc_qty</TD></TR></TABLE></TD><TD
PORT="alloc_qty.type" ALIGN="LEFT">decimal[12,4]</TD></TR>

      <TR><TD PORT="avail_qty" COLSPAN="2" ALIGN="LEFT"><TABLE BORDER="0"
CELLSPACING="0" ALIGN="LEFT"><TR ALIGN="LEFT"><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="15" HEIGHT="16"></TD><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="84" HEIGHT="16">avail_qty</TD></TR></TABLE></TD><TD
PORT="avail_qty.type" ALIGN="LEFT">decimal[12,4]</TD></TR>

      <TR><TD PORT="person_code" COLSPAN="2" ALIGN="LEFT"><TABLE BORDER="0"
CELLSPACING="0" ALIGN="LEFT"><TR ALIGN="LEFT"><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="15" HEIGHT="16"></TD><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="84"
HEIGHT="16">person_code</TD></TR></TABLE></TD><TD PORT="person_code.type"
ALIGN="LEFT">char[8]</TD></TR>

      <TR><TD PORT="partial_flag" COLSPAN="2" ALIGN="LEFT"><TABLE
BORDER="0" CELLSPACING="0" ALIGN="LEFT"><TR ALIGN="LEFT"><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="15" HEIGHT="16"></TD><TD ALIGN="LEFT"
FIXEDSIZE="TRUE" WIDTH="84"
HEIGHT="16">partial_flag</TD></TR></TABLE></TD><TD PORT="partial_flag.type"
ALIGN="LEFT">char[1]</TD></TR>

      <TR><TD ALIGN="LEFT" BGCOLOR="#ffffff">&lt; 1</TD><TD ALIGN="RIGHT"
BGCOLOR="#ffffff">0 rows</TD><TD ALIGN="RIGHT" BGCOLOR="#ffffff">0
&gt;</TD></TR>

    </TABLE>>

    URL="reqbackord.html"

    target="_top"



1Gooltip="reqbackord"

  ];

}



*De :* Nils Petzäll [mailto:notifications@github.com]
*Envoyé :* mardi 8 janvier 2019 18:11
*À :* schemaspy/schemaspy <schemaspy@noreply.github.com>
*Cc :* begooden-it <eric.vercelletto@begooden-it.com>; Mention <
mention@noreply.github.com>
*Objet :* Re: [schemaspy/schemaspy] SchemaSpy with Informix 12.10 FC10:
"cursor has been previously released" exception. SchemaSpy crashes (#424)



You could check the content of
/home/StagingArea/schemaSpy/kandoodb/diagrams/tables/reqbackord.implied1degrees.dot
something with label in company. If it's not secret or confidential you
could post the content and I can have look.

For the cell size, I wonder if there is a row content that we don't take
into account. It's probably the longest. I'll glance over the code for the
cell thing.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/schemaspy/schemaspy/issues/424#issuecomment-452377443>,
or mute the thread

**Comment by brian-hughes** [comment]
The problem here with the cursor already released comes from some versions of the Informix JDBC driver. Quick fix is to upgrade to the latest 4.50.1 driver. It's out on maven.

```
<!-- https://mvnrepository.com/artifact/com.ibm.informix/jdbc -->
<dependency>
    <groupId>com.ibm.informix</groupId>
    <artifactId>jdbc</artifactId>
    <version>4.50.1</version>
</dependency>
```

In detail. The 4.10.X drivers introduced a statement cleaner thread that runs every 15 seconds and looks for resources that need to be closed as they have no more references and could be garbage collected. The problem, then compounds as a couple JDBC metdata calls incorrectly failed to close resources causing this cleaner thread to close resources in the middle of the work sometimes. Hence some people see it often, some never. You can work around it with 4.10.JC12 where we added a connection flag to disable the cleaner thread (IFMXCONNECTION_CLEANER_THREADS=0). Or the best option is to upgrade to the newest 4.50.1 driver which has this issue properly fixed.

**Comment by npetzall** [comment]
@begooden-it what @brian-hughes says =)

Thanks @brian-hughes     

@begooden-it I'll close this and if you feel that @brian-hughes answer doesn't apply, please re-open.

**Comment by begooden-it** [comment]
Hi

I have no more the issue on the snapshot, with 14.10 embedded driver (4.50FC1)
